subprojects/logging/src/main/java/org/gradle/internal/logging/sink/GroupingProgressLogEventGenerator.java

```diff
@@ -84,6 +103,19 @@ private void onStart(ProgressStartEvent startEvent) {
             if (isGroupedOperation(startEvent.getBuildOperationCategory())) {
                 String header = startEvent.getLoggingHeader() != null ? startEvent.getLoggingHeader() : startEvent.getDescription();
                 operationsInProgress.put(buildOpId, new OperationGroup(startEvent.getCategory(), header, startEvent.getTimestamp(), startEvent.getBuildOperationId()));
+
+                if (future == null || future.isCancelled()) {
+                    future = executor.scheduleWithFixedDelay(new Runnable() {
+                        @Override
+                        public void run() {
+                            for (OperationGroup group : operationsInProgress.values()) {
+                                if ((group.lastUpdateTime + LONG_RUNNING_TASK_OUTPUT_FLUSH_TIMEOUT) < timeProvider.getCurrentTime()) {
+                                    group.flushOutput();
+                                }
+                            }
+                        }
+                    }, LONG_RUNNING_TASK_OUTPUT_FLUSH_TIMEOUT, 500, TimeUnit.MILLISECONDS);
```

Comment:
The initial delay is the second argument (`LONG_RUNNING_TASK_OUTPUT_FLUSH_TIMEOUT`). It will check if the 5s is passed every 500ms. Worst case scenario is 5500ms.